torchmultimodal/utils/common.py

```diff
@@ -121,13 +121,6 @@ def tensor_slice(x: Tensor, begin: List[int], size: List[int]) -> Tensor:
     return x[slices]
 
 
-def transpose_for_scores(
-    num_attention_heads: int, attention_head_size: int, x: Tensor
-) -> Tensor:
-    x = x.unflatten(-1, (num_attention_heads, attention_head_size))
-    return x.permute(0, 2, 1, 3)
```

Comment:
why do we need to replace the simple permute logic with the shift dim logic?

Reply:
shift_dim has built in error checking, allows use of negative dimensions, and calls `contiguous()` after permuting which is important if you call `view()` right after. so although it's replacing a single line with slightly more code it's a lot more flexible to use.